### PR TITLE
co-locate overloads of std.algorithm.mutation.swap()

### PR DESCRIPTION
**Pull request by WalterBright**
The two overloads are widely separated. This puts them together for easier study. The changes are mostly indenting.

Also has its own error message rather than the compiler generated one.

### CODE REVIEW

**Comment by WalterBright** [comment]
Gave up on the own message and the merge. Just co-located them.

**Comment by WalterBright** [comment]
dstyle fails with no message.

File: std/algorithm/mutation.d [hunk 1]
```diff
@@ -2837,6 +2837,12 @@ Params:
     lhs = Data to be swapped with `rhs`.
     rhs = Data to be swapped with `lhs`.
 */
+void swap(T)(ref T lhs, ref T rhs)
+    if (is(typeof(lhs.proxySwap(rhs))))
+{
+    lhs.proxySwap(rhs);
+}
+
 void swap(T)(ref T lhs, ref T rhs) @trusted pure nothrow @nogc
```
Comment (dkorpel) [inline std/algorithm/mutation.d:2846]: This overload is undocumented now

```suggestion
/// ditto
void swap(T)(ref T lhs, ref T rhs) @trusted pure nothrow @nogc
```

File: std/algorithm/mutation.d [hunk 2]
```diff
@@ -2837,6 +2837,13 @@ Params:
     lhs = Data to be swapped with `rhs`.
     rhs = Data to be swapped with `lhs`.
 */
+void swap(T)(ref T lhs, ref T rhs)
+    if (is(typeof(lhs.proxySwap(rhs))))
```
Comment (RazvanN7) [inline std/algorithm/mutation.d:2841]: D-scanner wants the if to be aligned with the function declaration.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**